build-system/pr-check/ci-job.js

```diff
@@ -27,9 +27,44 @@ const {
   setLoggingPrefix,
 } = require('../common/logging');
 const {determineBuildTargets} = require('./build-targets');
-const {isPullRequestBuild} = require('../common/ci');
+const {isCircleciBuild, isPullRequestBuild} = require('../common/ci');
 const {red} = require('../common/colors');
 const {updatePackages} = require('../common/update-packages');
+const {Worker} = require('worker_threads');
+
+/** @type {Worker} */
+let fastFailWorker;
+
+/**
+ * Starts the fast fail background polling service as a worker.
+ * @return {Promise<void>}
+ */
+function startFastFailPollingWorker() {
+  return new Promise((resolve, reject) => {
+    if (fastFailWorker) {
+      reject('The worker already exists');
+      return;
+    }
+    fastFailWorker = new Worker('./build-system/common/fast_fail.js', {
+      workerData: {
+        pid: process.pid,
+      },
+    });
+    fastFailWorker.on('message', (message) => {
+      logWithoutTimestamp(message);
+      resolve();
+    });
+  });
+}
+
+/**
+ * Stops the fast fail background process.
+ */
+function stopFastFailWorker() {
+  if (fastFailWorker) {
+    fastFailWorker.terminate();
+  }
+}
```

Comment:
For maintainability, can we move all this code into a new file? Maybe `build-system/common/fast-fail-worker.js`?

Reply:
Definitely